Library/ViewModels/PostCampaignCheckoutViewModel.swift

```diff
@@ -179,37 +162,37 @@ public class PostCampaignCheckoutViewModel: PostCampaignCheckoutViewModelType,
       .combineLatest(
         checkoutId,
         selectedCard,
-        paymentIntentClientSecretForExistingCards,
-        storedCardsValues
+        paymentIntentClientSecretForExistingCards
       )
 
     // Runs validation for pre-existing cards that were created with setup intents originally but require payment intents for late pledges.
     let validateCheckoutExistingCard = validateCheckoutExistingCardInput
       .takeWhen(self.submitButtonTappedProperty.signal)
-      .switchMap { checkoutId, selectedExistingCreditCard, paymentIntentClientSecret, storedCards in
-        let selectedStoredCard = storedCards.first { $0.id == selectedExistingCreditCard.savedCreditCardId }
-        let selectedCardStripeCardId = selectedStoredCard?.stripeCardId ?? ""
+      .switchMap { checkoutId, selectedCard, paymentIntentClientSecret in
+
+        assert(
+          selectedCard.stripePaymentMethodId != nil,
```

Comment:
nit: Maybe check that the string is non-empty here instead of just non-nil (just in case we default to empty string ever in the future)?